src/Symfony/Component/Workflow/Marking.php

```diff
@@ -27,23 +27,58 @@ class Marking
     public function __construct(array $representation = [])
     {
         foreach ($representation as $place => $nbToken) {
-            $this->mark($place);
+            $this->mark($place, $nbToken);
         }
     }
 
-    public function mark(string $place): void
+    /**
+     * @param int $nbToken
+     */
+    public function mark(string $place /* , int $nbToken = 1 */): void
     {
-        $this->places[$place] = 1;
+        $nbToken = 1 < \func_num_args() ? func_get_arg(1) : 1;
+
+        if ($nbToken < 1) {
+            throw new \InvalidArgumentException(sprintf('The number of tokens must be greater than 0, "%s" given.', $nbToken));
+        }
+        dump(isset($this->place[$place]));
+        if (!\array_key_exists($place, $this->places)) {
+            $this->places[$place] = 0;
+        }
+        $this->places[$place] += $nbToken;
     }
 
-    public function unmark(string $place): void
+    /**
+     * @param int $nbToken
+     */
+    public function unmark(string $place /* , int $nbToken = 1 */): void
     {
-        unset($this->places[$place]);
+        $nbToken = 1 < \func_num_args() ? func_get_arg(1) : 1;
+
+        if ($nbToken < 1) {
+            throw new \InvalidArgumentException(sprintf('The number of tokens must be greater than 0, "%s" given.', $nbToken));
+        }
+
+        if (!$this->has($place)) {
+            throw new \InvalidArgumentException(sprintf('The place "%s" is not marked.', $place));
+        }
+
+        $nbToken = $this->places[$place] - $nbToken;
+
+        if (0 > $nbToken) {
+            throw new \InvalidArgumentException(sprintf('The place "%s" could not contain a negative token number.', $place));
```

Comment:
I suggest rewording the message to say that you cannot unmark the place from more token than its marking, perhaps mentioning the existing marking number and the number of removed token. It might make the error easier to understand.